Dockerfile.production

```diff
@@ -0,0 +1,114 @@
+    apt-get update -qq && \
+    echo 'en_US.UTF-8 UTF-8' > /etc/locale.gen && \
+    apt-get install -y apt-utils curl locales && \
+    apt-get upgrade -y && apt-get clean && \
+    rm -rf /var/lib/apt/lists/*"
+
+# Sets an environment variable with the bundle directory
+ENV LANG=en_US.UTF-8
+
+# Generates localisation files from templates
+RUN locale-gen
+
+# Install NODE_MAJOR
+ARG NODE_MAJOR=14
+RUN bash -c "export DEBIAN_FRONTEND=noninteractive && \
+    curl -sL https://deb.nodesource.com/setup_$NODE_MAJOR.x | bash - && \
+    apt-get update && apt-get install -y nodejs"
+
+# Install Yarn via Debian package repository
+RUN bash -c "export DEBIAN_FRONTEND=noninteractive && \
+    curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - && \
+    echo 'deb https://dl.yarnpkg.com/debian/ stable main' | tee /etc/apt/sources.list.d/yarn.list && \
+    apt-get update -qq && apt install yarn"
+
+# Install Bundler
+ARG BUNDLER_VERSION=2.4.6
+RUN bash -c "gem update --system && gem install bundler -v $BUNDLER_VERSION"
+
+# Base image for building dependencies
+FROM base_image AS build_image
+
+# Run security updates, install build-essential git-core
+RUN bash -c "export DEBIAN_FRONTEND=noninteractive && \
+    apt-get update -qq && \
+    apt-get upgrade -y && \
+    apt-get install -y build-essential git-core && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*"
+
+# Build image for ruby dependencies
+FROM build_image AS gems
+
+# Create a directory for our application
+# and set it as the working directory
+WORKDIR /app
+
+# Add our Gemfile
+COPY Gemfile Gemfile.lock .ruby-version /app/
+
+# Install gems
+RUN bash -c "bundle config set --local without 'development test' && \
+  bundle config set --local deployment 'true' && \
+  bundle install --no-cache && \
+  bundle clean --force"
+
+# Build image for npm dependencies
+FROM gems AS packages
+
+# Create a directory for our application
+# and set it as the working directory
+WORKDIR /app
+
+# Add Javascript files
+COPY package.json yarn.lock /app/
+
+RUN bash -c "bundle exec yarn install"
+
+# Build image for application
+FROM packages AS application
+
+# Create a directory for our application
+# and set it as the working directory
+WORKDIR /app
+
+RUN groupadd -r app && \
+    useradd --no-log-init -r -g app -d /app app
+
+# Create various directories
+RUN mkdir /app/log && \
+    mkdir /app/tmp && \
+    chown -R app:app /app
+
+# Copy neccessary files to app directory
+COPY --chown=app:app ./app/ /app/app/
+COPY --chown=app:app ./bin/ /app/bin/
+COPY --chown=app:app ./config/ /app/config/
+COPY --chown=app:app ./config.ru /app/config.ru
+COPY --chown=app:app ./lib/ /app/lib/
+COPY --chown=app:app ./public/ /app/public/
+COPY --chown=app:app ./Rakefile /app/Rakefile
+COPY --chown=app:app ./babel.config.js /app/babel.config.js
+COPY --chown=app:app ./postcss.config.js /app/postcss.config.js
+
+COPY --from=packages --chown=app:app /app/node_modules/ /app/node_modules
```

Comment:
I think because of the way the dockerfile is written, lines 101–102 aren't necessary. It's copying from a previous stage, but this stage already inherits from those stages. We could probably just remove them.

The situation in which they'd be beneficial is if the `application` stage inherited from a different stage, in which case we could copy only the build artefacts and leave behind any tempfiles or build dependencies. But we're not doing that here.